packages/admin-next/admin-sdk/src/lib/config.ts

```diff
@@ -59,6 +59,14 @@ export async function getViteConfig(
       }),
     ],
   }
+
+  if (options.vite) {
+    const customConfig = options.vite(baseConfig)
+
+    return mergeConfig(baseConfig, customConfig)
```

Comment:
Don't think its realistic to validate what is passed, as they the config is very deep (it has almost all rollup and esbuild options). When we get the `defineConfig` the type will be there to help guide people. Also Vite won't crash if you return an invalid config, it will just discard it.

Reply:
> Also Vite won't crash if you return an invalid config, it will just discard it.

Cool, sounds good